### PR TITLE
Fix git_clone

### LINKED ISSUE #16: Bug with updating from GIT
I've just tried to obtain a module via its git URL. The resulting ZIP does not match the one provided by upstream; it seems some directory-separator is missing:

image image

The left screenshot shows the structure of the ZIP as provided by the module's author. The right one is what repo-util created. Note the files:

* `commonfunctions.sh` in the original module is `common/functions.sh`
* `systemplaceholder` originally is `system/placeholder`

No files are missing. Is there some bug with `_get_module_from_git`? Or rather with `git_clone`?

```python
    with ZipFile(out, "w", ZIP_DEFLATED) as f:
        for dir_path, dir_names, file_names in os.walk(repo_dir):
            file_path = dir_path.replace(repo_dir.as_posix(), "")
            for file_name in file_names:
                f.write(os.path.join(dir_path, file_name), file_path + file_name)
```

The last line looks wrong. Shouldn't that be

```python
               f.write(os.path.join(dir_path, file_name), os.path.join(file_path,file_name))
```

Adjusting it that way makes the archive looking fine. Further, the two lines before above block:

```python
    shutil.rmtree(repo_dir.joinpath(".git"), ignore_errors=True)
    shutil.rmtree(repo_dir.joinpath(".github"), ignore_errors=True)
```

miss some more removals: `.gitlab` tree, and files like `.gitattributes` and `.gitignore`.

### PR DESCRIPTION
**Pull request by IzzySoft**
fixes #16 (details described there)

### CODE REVIEW

**Comment by IzzySoft** [comment]
Oops, I should stick to the coding style – fixed that. Feel free to squash :wink:

File: sync/File.py [hunk 1]
```diff
@@ -90,12 +90,17 @@ def git_clone(url: str, out: Path):
 
     shutil.rmtree(repo_dir.joinpath(".git"), ignore_errors=True)
     shutil.rmtree(repo_dir.joinpath(".github"), ignore_errors=True)
+    shutil.rmtree(repo_dir.joinpath(".gitlab"), ignore_errors=True)
+    if os.path.exists(repo_dir.joinpath(".gitattributes")):
```
Comment (SanmerDev) [inline sync/File.py:94]:
```suggestion
    if repo_dir.joinpath(".gitattributes").exists():
```

**Comment by SanmerDev** [comment]
All files starting with `.git` are related to git, we should rewrite the cleaner.

**Comment by SanmerDev** [comment]
- packaging method is changed to `shutil.make_archive`
- git file cleanup method changed, all files and folders starting with `.git` will be cleared

These changes will be committed later, I'm sorry this pr won't be merged 😶‍🌫️.

**Comment by IzzySoft** [comment]
Thanks for fixing, @ya0211

> I'm sorry this pr won't be merged :face_in_clouds:.

I'm getting used to your always just closing any PR I open and never merging them :see_no_evil: If I should rather refrain from opening PRs and instead simply keep my suggestions in the issues, just let me know – and I save myself the trouble of maintaining a fork for that :wink: (no bad feelings, I just need to know)

> All files starting with `.git` are related to git, we should rewrite the cleaner.

Is this a fact, or an assumption? Sure it would be really bad practice having non-git-related files/directories starting with `.git`, and I would never even consider doing so – but can we be sure nobody does? The ones named were clear candidates. And directories like `.github` or `.gitlab` are not directly related to git itself but to the corresponding "forges" (just that their names also start with "Git"); I need to check, but IIRC there's also `.gitea` (again matching the pattern, and again specific to a forge and thus no problem) and probably `.forgejo` etc for other forges (which might need to be added later explicitly should we encounter them). I've never seen any `.git*` files which would belong to the code maintained in a repo. But can we go so far as using a glob here? Are we really sure? Or might there be an exception one day causing head-aches?

No complaints, just want to make sure.

**Comment by SanmerDev** [comment]
It's not always to close your pr🤪, I have a new idea that is too different from old one, so it's meaningless to merge it. Most of the time, it's ok to let me fix the bug 😉.
In fact, these files will not affect the installation of the module. The only `.git` has a huge impact on the size of the zipFile. Let’s deal with it list that for now 🤔.

**Comment by IzzySoft** [comment]
> It's not always to close your pr

Haha, no, I get that. I saw the changes you applied finally, and they were quite different from my "simple fixes". You have the "broader image" in mind, I only see spots. Just looks a bit strange when it happens the second times from 2 tries :zany_face: I could of course ask first if I shall open a PR with the proposed solution.

> only `.git` has a huge impact on the size of the zipFile.

That's usually true. A list of "known ones" (maybe an array to be walked in a for-loop?) couldn't hurt, though. I don't expect any module repo having a huge collection of Github actions (or GitLab CI definitions) set up, but who knows? It could still easily double the size of some less-than-10k modules. If you'd consider such a list, I guess the most frequent candidates have been named already: `\.git(hub|lab|attributes|ignore|ea)` for a quick match :wink:

**Pull request closed without merging**